modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/upload/DLUploadFileEntryHandler.java

```diff
@@ -109,6 +128,69 @@ private boolean _exists(long groupId, long folderId, String fileName) {
 		}
 	}
 
+	private Map<String, Serializable> _getCustomFieldsMap(
```

Comment:
Why do we need this custom logic here? Why not use the same approach as ServiceContextFactory?

```
Map<String, Serializable> expandoBridgeAttributes =
	PortalUtil.getExpandoBridgeAttributes(
		ExpandoBridgeFactoryUtil.getExpandoBridge(
			serviceContext.getCompanyId(), className),
		httpServletRequest);
```

Reply:
If there's a difference in the format (JSON vs regular expandos), I have a problem with putting this logic in the generic uploader. It looks to me that this case only affects WC, so it should be fixed in Journal, not in DM (e.g. with a custom upload handler).